scripts/fast-reboot

```diff
@@ -70,44 +88,83 @@ docker exec -i bgp killall -9 bgpd
 docker kill lldp > /dev/null
 
 # Kill teamd, otherwise it gets down all LAGs
+# Note: teamd must be killed before syncd, because it will send the last packet through CPU port
+# TODO: stop teamd gracefully to allow teamd to send last valid update to be sure we'll have 90 seconds reboot time
 docker kill teamd > /dev/null
 
+if [[ "$REBOOT_TYPE" = "warm-reboot" ]]; then
+    # Freeze orchagent for warm restart
+    docker exec -i swss /usr/bin/orchagent_restart_check || {
+        echo "RESTARTCHECK failed\n" >&2
+        # TODO: remove this sleep if we have better timing control
+        sleep 1
```

Comment:
I understand your concern about script name. It was named as it because it includes many features besides freezing. So the comment is added just to clarify the purpose. Both the code and the comment are correct and serve the intended purpose.

I will update TODO.

---
In reply to: [225441284](https://github.com/Azure/sonic-utilities/pull/338#discussion_r225441284) [](ancestors = 225441284)